### PR TITLE
Tailor output to look better on a GitHub releases page

### LINKED ISSUE #87: Add templating that covers github release page features
Change the user template for the changelogs generated for github to use `@user` instead of `[user](profile)` to get formatted results in the releases page that annotates individual contributors:

Screenshot 2023-06-28 at 9 18 34 AM

There are probably more features to look at here, this is just an example. To cover these kinds of things it would be nice to be able to specify a name for a format that is tailored for this kind of output:

```bash
$ chronicle ... -o github-release

# get markdown that is targeted for github release pages
```

### PR DESCRIPTION
**Pull request by wagoodman**
This PR tries to clean up the markdown format and go for a more minimal appearance:

**Today:**

<kbd>Screenshot 2023-09-12 at 2 55 03 PM</kbd>

**With changes:** (and `title` set to empty string)

<kbd>Screenshot 2023-09-14 at 2 37 15 PM</kbd>

Note this removes the version / vcs url / date from the top lines of the output. What's really happening is that the `title` has been set to `""`, however, it would default to `{{ .Version }}` if nothing is provided. I wanted to remove the top lines since the github releases page already has this information, for example:

<kbd>Screenshot 2023-09-14 at 2 38 42 PM</kbd>

...with the top fields removed it reads a lot better. However, there may be users that are depending on the version being somewhere in the top of the document. For that reason the title now allows for templating and defaults to `{{ .Version }}`.

Detailed changes:
- if there is no title, do not render it (leaves out a since `#` at the top of the document)
- allow for templating in the user title and set the default title to the `.Version` value
- condense link annotations by removing square brackets and removing unnecessary words such as `Issue` and `PR`
- adds all reference links in a single parenthetical statement
- add `@` in front of usernames (which the releases page should incorporate in a "contributors" footer)
- ensure each changelog entry does not end in punctuation. This is to separate out the change description from the references.
- full changelog link has been moved to the bottom of the document since it allows for a deeper, more raw dive into the changes, and leaves the summary of changes to flow to the top of the document.

Fixes #87

### CODE REVIEW

**Comment by wagoodman** [comment]
Failing CI will be fixed by https://github.com/anchore/chronicle/pull/105 (I'm using go 1.21.1 locally, so go mod tidy really messed with things here)

File: chronicle/release/format/markdown/presenter.go [hunk 1]
```diff
@@ -71,7 +82,7 @@ func (m Presenter) formatChangeSections(changes change.Changes) string {
 			result += formatChangeSection(section.Title, summaries) + "\n"
 		}
 	}
-	return result
+	return strings.TrimRight(result, "\n")
```
Comment (willmurphyscode) [inline chronicle/release/format/markdown/presenter.go:85]: Why do we now need to remove trailing newlines?

File: chronicle/release/format/markdown/presenter.go [hunk 2]
```diff
@@ -83,14 +94,41 @@ func formatChangeSection(title string, summaries []change.Change) string {
 }
 
 func formatSummary(summary change.Change) string {
-	result := fmt.Sprintf("- %s", summary.Text)
+	result := fmt.Sprintf("- %s", strings.TrimSpace(summary.Text))
+	if !endsWithPunctuation(result) {
+		switch result[len(result)-1:] {
+		case "!", ".", "?":
```
[the remark this answers is not in the document]
Comment (wagoodman) [reply chronicle/release/format/markdown/presenter.go:100]: oversight in refactor, it's not needed. I'll adjust

File: chronicle/release/format/markdown/presenter.go (hunk 1 above)
Comment (wagoodman) [reply chronicle/release/format/markdown/presenter.go:85]: Now there is a footer and it became noticeable that we have a large amount of whitespace before the footer -- this makes the whitespace between the content and footer smaller

**Review by willmurphyscode — APPROVED** [review]

**Pull request merged**